pdr_backend/predictoor/base_predictoor_agent.py

```diff
@@ -80,6 +81,7 @@ def take_step(self):
         for addr in self.feeds:
             self._process_block_at_feed(addr, block["timestamp"])
 
+    @retry_function(5, 2, disallowed_exceptions=())
```

Comment:
This is still way too specific.

- What if we have a bug in the implementation of _process_block_at_feed()? 
- It's highly possible since that's big

What we _should_ be doing: detect the specific network issue or two, and handle that very specifically. So that other errors that we don't anticipate _do_ pop out, and we properly address them. Nothing general.

Finally: we will want a unit test for that specific error handing.